### DISCUSSION TITLE
Dropping support for old node versions in ansible-language-server

### DISCUSSION THREAD

**Discussion by ssbarnea** [ansible/ansible-language-server#355] [Polls]
Current versions of vscode use node 16 and the is way an extension can control that.

Also running multiple pipelines for ancient node versions makes no sense, especially when we do not have a specific test for keeping compatibility with a specific consumer.

That is why I am proposing removing node `12` and node `14` from testing pipelines and testing only with node `16` and current latest `stable`, so we know when a newer version will introduce breaking changes.

If someone needs to keep compatibility with older node version because their editor cannot use the same version as vscode, they will have to make a pull request that adds tests for this.

We do want to allow others to write ALS clients for other editors, but in absence of tests we cannot give any kind of warranties that even a minor change to it will not break their use case.

@yaegassy That might interest you in particular.

**Comment by yaegassy** [comment]
@ssbarnea Voted. :)
